### PR TITLE
[FIX] point_of_sale: fix traceback when fast click validate

### PR DESCRIPTION
**Pull request by robinengels**
Current behavior:
When clicking multiple times on the validate button in PoS
a traceback appears.
When creating a customer from the PoS, if you click mutliple times
on the create button the customer would be created multiple times
in the Db.

Steps to reproduce issue 1:
- To reproduce you need a slow internet connection, you
  can do that in the developper menu of chrome (F12) and
  in the network tab select 'slow 3G' instead of 'no throttling'
- Go in a PoS session, make an order
- Go in the payment screen
- Press validate multiple times.
- A traceback appears

Steps to reproduce issue 2:
- Go in the customer selection screen
- Go in the create customer screen
- Enter a name, and click multiple times on the create button
- If you check customer list the new customer should appear
  multiple times

opw-2813512
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by robinengels** [comment]
@pimodoo

**Comment by caburj** [comment]
I would prefer having a state variable in the component that will serve a lock, such that if it's true, no need to execute the method. Practically similar to your solution but more principled (at least in my opinion). Also, I think you changed the wrong `saveChanges`. The right one is in PartnerListScreen.

**Comment by robinengels** [comment]
@caburj this should be better

File: addons/point_of_sale/static/src/js/Screens/PaymentScreen/PaymentScreen.js [hunk 1]
```diff
@@ -178,6 +186,7 @@ odoo.define('point_of_sale.PaymentScreen', function (require) {
                 }
                 await this._finalizeValidation();
             }
+            this.validateOrderIsBlocked = false;
```
Comment (caburj) [inline addons/point_of_sale/static/src/js/Screens/PaymentScreen/PaymentScreen.js:189]: `this.state`. Also, put it in try-finally statement.

**Review by caburj — APPROVED** [review]

**Comment by robinengels** [comment]
@robodoo r+

**Pull request closed without merging**